File: backend/models/message.py
from pymongo import MongoClient
from bson.objectid import ObjectId
import os
from datetime import datetime

# MongoDB connection
client = MongoClient(os.getenv("MONGODB_URI"))
db = client[os.getenv("MONGODB_DB_NAME", "sportmatch")]

class Message:
    @staticmethod
    def get_by_match_id(match_id):
        return list(db.messages.find({"matchId": match_id}).sort("timestamp", 1))
    
    @staticmethod
    def create(message_data):
        message_data["timestamp"] = datetime.now()
        message_data["createdAt"] = datetime.now()
        message_data["updatedAt"] = datetime.now()
        message_data["read"] = False
        
        result = db.messages.insert_one(message_data)
        message_data["_id"] = result.inserted_id
        
        return message_data
    
    @staticmethod
    def mark_as_read(match_id, user_id):
        result = db.messages.update_many(
            {
                "matchId": match_id,
                "receiverId": user_id,
                "read": False
            },
            {
                "$set": {
                    "read": True,
                    "updatedAt": datetime.now()
                }
            }
        )
        
        return result.modified_count

File: models/Message.ts
import type { ObjectId } from "mongodb"
import clientPromise from "@/lib/mongodb"

export interface Message {
  _id?: ObjectId
  matchId: string
  senderId: string
  receiverId: string
  text: string
  timestamp: Date
  read: boolean
  createdAt?: Date
  updatedAt?: Date
}

export async function getMessagesByMatchId(matchId: string): Promise<Message[]> {
  try {
    const client = await clientPromise
    const db = client.db(process.env.MONGODB_DB_NAME || "sportmatch")

    return db.collection("messages").find({ matchId }).sort({ timestamp: 1 }).toArray() as Promise<Message[]>
  } catch (error) {
    console.error("Error in getMessagesByMatchId:", error)
    return []
  }
}

export async function createMessage(messageData: Omit<Message, "_id" | "createdAt" | "updatedAt">): Promise<Message> {
  const client = await clientPromise
  const db = client.db(process.env.MONGODB_DB_NAME || "sportmatch")

  const now = new Date()
  const message = {
    ...messageData,
    createdAt: now,
    updatedAt: now,
  }

  const result = await db.collection("messages").insertOne(message)

  return {
    ...message,
    _id: result.insertedId,
  }
}

export async function markMessagesAsRead(matchId: string, userId: string): Promise<void> {
  try {
    const client = await clientPromise
    const db = client.db(process.env.MONGODB_DB_NAME || "sportmatch")

    await db.collection("messages").updateMany(
      {
        matchId,
        receiverId: userId,
        read: false,
      },
      {
        $set: {
          read: true,
          updatedAt: new Date(),
        },
      },
    )
  } catch (error) {
    console.error("Error in markMessagesAsRead:", error)
  }
}
